src/dmd/lexer.d

```diff
@@ -288,14 +290,34 @@ class Lexer : ErrorHandler
         }
     }
 
+    /// Returns: a newly allocated `Token`.
+    Token* allocateToken() pure nothrow @safe
+    {
+        if (tokenFreelist)
+        {
+            Token* t = tokenFreelist;
+            tokenFreelist = t.next;
+            t.next = null;
+            return t;
+        }
+        return new Token();
+    }
+
+    /// Frees the given token by returning it to the freelist.
+    void free(Token* token) pure nothrow @nogc @safe
```

Comment:
Rename this to `freeToken` for consistency with `allocateToken` and to remove possible confusion with C's `free`.

Reply:
Renamed to `releaseToken`.